File: 168.excel-sheet-column-title.py
#
# @lc app=leetcode id=168 lang=python3
#
# [168] Excel Sheet Column Title
#

# @lc code=start
class Solution:
    def convertToTitle(self, n: int) -> str:
        n = n - 1
        numbers = []
        while not n//26 == 0: # in this way it will be easy to understand
            numbers.append(n // 26)
            n = n // 26
        numbers.append(n)
        print(numbers)
        
        return "".join([ chr(ord('A') + number - 1) for number in numbers])
    
# @lc code=end

if __name__ == "__main__":
    print(Solution().convertToTitle(26))
    print(Solution().convertToTitle(30))

File: golang/168.excel-sheet-column-title.go
/*
 * @lc app=leetcode id=168 lang=golang
 *
 * [168] Excel Sheet Column Title
 */
package main

import (
	"fmt"
	"strings"
)

// @lc code=start
func convertToTitle(columnNumber int) string {
	ret := []string{}
	columnNumber = columnNumber
	for columnNumber > 0 {
		ret = append(ret, string([]byte{byte(columnNumber%26 + 'A')}))
		columnNumber /= 26
	}
	return strings.Join(ret, "")
}

// @lc code=end
func main() {
	fmt.Println(convertToTitle(26))
	fmt.Println(convertToTitle(27))
	fmt.Println(convertToTitle(52))
	fmt.Println(convertToTitle(701))
}
